android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt

```diff
@@ -374,11 +386,15 @@ class MusicService : HeadlessJsTaskService() {
         scope.launch {
             event.notificationStateChange.collect {
                 when (it) {
-                    is NotificationState.POSTED -> startForeground(
-                            it.notificationId,
-                            it.notification
-                    )
-                    is NotificationState.CANCELLED -> stopForeground(true)
+                    is NotificationState.POSTED -> {
+                        if (!isForegroundService) {
```

Comment:
non-blocking: What does this `isForegroundService` flag do? Makes sure we only start once? Then we should use a different name like `hasStartedForegroundService` or something.